test/teos/conftest.py

```diff
@@ -0,0 +1,199 @@
+import os
+import pytest
+import random
+import subprocess
+from time import sleep
+from os import makedirs
+from shutil import rmtree, copy
+from decimal import Decimal, getcontext
+
+from teos.teosd import get_config
+from teos.utils.auth_proxy import AuthServiceProxy, JSONRPCException
+
+
+getcontext().prec = 10
+utxos = list()
+btc_addr = None
+
+
+cmd_args = {"BTC_NETWORK": "regtest"}
+config = get_config(cmd_args, ".teos")
+
+bitcoin_cli = AuthServiceProxy(
+    "http://%s:%s@%s:%d"
+    % (
+        config.get("BTC_RPC_USER"),
+        config.get("BTC_RPC_PASSWORD"),
+        config.get("BTC_RPC_CONNECT"),
+        config.get("BTC_RPC_PORT"),
+    )
+)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def prng_seed():
+    random.seed(0)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def run_bitcoind(dirname=".test_bitcoin"):
+    # Run bitcoind in a separate folder
+    makedirs(dirname, exist_ok=True)
+
+    bitcoind = os.getenv("BITCOIND", "bitcoind")
+
+    copy(os.path.join(os.path.dirname(__file__), "bitcoin.conf"), dirname)
+    subprocess.Popen([bitcoind, f"--datadir={dirname}"])
+
+    # Generate some initial blocks
+    setup_node()
+    yield
+
+    bitcoin_cli.stop()
+    rmtree(dirname)
+
+
+def setup_node():
+    global btc_addr
+
+    # Check bitcoind is running while generating the address
+    while True:
+        # FIXME: Not creating a new bitcoin_cli here creates one of those Request-Sent errors I don't know how to fix
+        #        Not a big deal, but it would be nicer not having to.
+        bitcoin_cli = AuthServiceProxy(
+            "http://%s:%s@%s:%d"
+            % (
+                config.get("BTC_RPC_USER"),
+                config.get("BTC_RPC_PASSWORD"),
+                config.get("BTC_RPC_CONNECT"),
+                config.get("BTC_RPC_PORT"),
+            )
+        )
+        try:
+            btc_addr = bitcoin_cli.getnewaddress()
+            break
+
+        except ConnectionError:
+            sleep(1)
+        except JSONRPCException as e:
+            if "Loading wallet..." in str(e):
+                sleep(1)
+
+    # Mine enough blocks so coinbases are mature and we have enough funds to run everything
+    bitcoin_cli.generatetoaddress(105, btc_addr)
+    create_initial_transactions()
+
+
+def create_initial_transactions(fee=Decimal(5 / pow(10, 5))):
```

Comment:
Not a big deal, but the default `fee` above results in `Decimal('0.0000500000000000000023960868011929647991564706899225711822509765625')` due to floating point encoding issues; maybe cleaner to just write `fee=Decimal('0.00005')`.